frontend/system/src/Admin/PendingChanges/Details.tsx

```diff
@@ -0,0 +1,189 @@
+import { useBreadcrumbs, useNotification } from '@common/hooks';
+import { CheckIcon, ChevronDownIcon, CloseIcon, EditIcon } from '@common/icons';
+import { useTranslation } from '@common/intl';
+import { UpsertEntityInput } from '@common/types';
+import {
+  BasicSpinner,
+  Box,
+  ButtonWithIcon,
+  LoadingButton,
+  NothingHere,
+} from '@common/ui';
+import { RouteBuilder } from '@common/utils';
+import { TreeView } from '@mui/lab';
+import { AppRoute } from 'frontend/config/src';
+import React, { useEffect, useState } from 'react';
+import { useNavigate, useParams } from 'react-router';
+import { useApprovePendingChange } from '../api';
+import { useEditPendingChange } from '../api/hooks/useEditPendingChange';
+import { useNextPendingChange } from '../api/hooks/useNextPendingChange';
+import { usePendingChange } from '../api/hooks/usePendingChange';
+import { useRejectPendingChange } from '../api/hooks/useRejectPendingChange';
+import { EditPendingChange } from './EditPendingChange';
+import { PendingChangeTreeItem } from './TreeItem';
+
+export const PendingChangeDetails = () => {
+  const { id } = useParams();
+  const { setSuffix } = useBreadcrumbs();
+  const t = useTranslation('system');
+  const navigate = useNavigate();
+  const { error, success } = useNotification();
+
+  const { mutateAsync: rejectPendingChange, isLoading: rejectionLoading } =
+    useRejectPendingChange();
+  const { mutateAsync: approvePendingChange, isLoading: approvalLoading } =
+    useApprovePendingChange();
+  const { mutateAsync: editChange, isLoading: editChangeLoading } =
+    useEditPendingChange();
+
+  const [isEditMode, setEditMode] = useState(false);
+
+  const [expanded, setExpanded] = useState<string[]>([]);
+  const [entity, setEntity] = useState<UpsertEntityInput | null>(null);
+
+  const { data: pendingChange, isLoading } = usePendingChange(id ?? '');
+  const next = useNextPendingChange(id ?? '');
+
+  // Set entity name in the breadcrumb
+  useEffect(() => {
+    if (pendingChange?.name) setSuffix(pendingChange.name);
+  }, [pendingChange?.name]);
+
+  // Parse entity from pending change body
+  useEffect(() => {
+    if (pendingChange) setEntity(JSON.parse(pendingChange.body));
+  }, [pendingChange]);
+
+  // Expand all nodes of the tree
+  useEffect(() => {
+    if (pendingChange) {
+      const expandedIds: string[] = [];
+
+      const addToExpandedIds = (ent?: UpsertEntityInput | null) => {
+        if (ent) {
+          const nodeId = ent.code || ent.description || '?';
+          expandedIds.push(nodeId);
+          ent.children?.forEach(addToExpandedIds);
+
+          ent?.properties && expandedIds.push(`${nodeId}_properties`);
+        }
+      };
+      addToExpandedIds(entity);
+
+      setExpanded(expandedIds);
+    }
+  }, [entity]);
```

Comment:
Is it better to have 2 use effects rather than combining into a single one? Or is entity update somewhere else?

I don't have an issue with it how it is was just wondering, as it looks like could be combined with the one above...

Reply:
Ah great catch - yeah the recommendation is to split up the useEffects with different dependencies, so the effect code only runs when it actually needs to... 

A previous iteration I had `entity` being updated more regularly, so separated that effect out from the one to run if the `pendingChange` changed... but now that code is gone, I've merged them back to one again (I think more efficient to queue all the state changes from one effect, so they are applied in one render) 

[1b9122f](https://github.com/msupply-foundation/unified-codes/pull/513/commits/1b9122f72c8106b859dec0c32c411f737d532d78)